src/states/node.rs

```diff
@@ -2203,24 +2209,17 @@ impl Node {
             self.merge_if_necessary();
 
             outbox.send_event(Event::Tick);
-            return true;
-        }
-
-        if self.rt_timer_token == Some(token) {
+        } else if self.rt_timer_token == Some(token) {
             self.rt_timeout = cmp::min(Duration::from_secs(RT_MAX_TIMEOUT_SECS),
                                        self.rt_timeout * 2);
             trace!("{:?} Scheduling next RT request for {} seconds from now.",
                    self,
                    self.rt_timeout.as_secs());
             self.rt_timer_token = Some(self.timer.schedule(self.rt_timeout));
-            if self.send_rt_request().is_err() {
-                return true;
-            }
+            self.send_rt_request().is_err();
```

Comment:
Probably `send_rt_request` and `send_candidate_approval` could have return type `()` now? Serialisation errors can just be logged inside them. Then this `is_err()` here could go.